devfile.yaml

```diff
@@ -0,0 +1,86 @@
+apiVersion: 1.0.0
+
+metadata:
+  generateName: che-plugin-registry-dev-
+
+
+components:
+
+  - mountSources: true
+    memoryLimit: 256M
+    type: dockerimage
+    image: sunix/git-devtools
+    alias: git
+
+  - alias: didact-plugin
+    type: chePlugin
+    id: redhat/vscode-didact/latest
+    preferences:
+      didact.openDefaultTutorialAtStartup: true
+      che.welcome.enable: false
+      didact.defaultUrl: /projects/che-plugin-registry/.che-didact-guide.md
+
+  - id: ms-vscode/vscode-github-pullrequest/latest
+    type: chePlugin
+
+  - mountSources: true
+    memoryLimit: 2G
+    type: dockerimage
+    image: 'quay.io/eclipse/che-plugin-registry-dev:latest'
+    alias: builder
+    env:
+      - value: '-m 1'
+        name: CONCURRENTLY_OPTS
+  - mountSources: true
+    endpoints:
+      - name: registry
+        port: 8080
+    memoryLimit: 256Mi
+    type: dockerimage
+    volumes:
+      - name: registryv3
+        containerPath: /usr/local/apache2/htdocs/v3
+    image: 'quay.io/eclipse/che-plugin-registry:nightly'
+    alias: eclipse-che-plugin-registry
+
+commands:
+
+  - name: 1. build
+    actions:
+      - workdir: '${CHE_PROJECTS_ROOT}'
+        type: exec
+        command: |
+          set -o xtrace;
+          rm -rf build;
+          mkdir build;
+          cp che-plugin-registry/build/scripts/meta.yaml.schema build/ &&
+          cp -rf che-plugin-registry/v3 build/v3 &&
+          ## generate_vscode_extension.sh
+          cd /projects/che-plugin-registry/tools/build &&
+          export SKIP_TEST=true SKIP_FORMAT=true SKIP_LINT=true  &&
+          yarn &&
+          yarn build &&
+          node -max-old-space-size=1024 lib/entrypoint.js --output-folder:/projects/build &&
+          ##
+          cd /projects/build &&
```

Comment:
it seems fragile as there is a build.sh script. probably need to enhance build.sh script or add a new script ?

Reply:
yes can we adapt the script to accept custom options (output folder etc ...) in another iteration?
ideally we should use the same single script in CI and in Che.